File: Problems_1-10/Problem4/Problem4.py
# Problem 4 :: Largest palindrome product

def isPalindromeNumber(n):
    n = str(n)
    return n == n[::-1]
    

if __name__ == "__main__":
    largestPalindrome = 0
    for i in range(900,1001):
        for j in range(900, 1001):
            if isPalindromeNumber(i*j) and largestPalindrome < (i*j): largestPalindrome = (i*j)
    print(largestPalindrome)

File: Problems_1-10/Problem4/Problem4.cpp
// Problem 4 :: Largest palindrome product

#include <bits/stdc++.h>

bool isPalindromeNumber(int n){
    int temp = n, flipped = 0, i = 0;
    while(temp != 0){
        int digit = temp % 10;
        temp /= 10;
        flipped = (flipped + digit) * 10;
    }
    if(flipped/10 == n) return true;
    return false;
}
    
int main(){
    int largestPalindrome = 0;
    for(int i=1000;i>=900;i--){
        for(int j=1000;j>=900;j--){
            if (i*j > largestPalindrome && isPalindromeNumber(i*j)) largestPalindrome = i*j;
            
        }
    }
    std::cout << largestPalindrome;
}
